openapi.yaml

```diff
@@ -0,0 +1,234 @@
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/inline_response_200'
+              examples:
+                0:
+                  value: |
+                    {"apiVersion":"1.0","data":[{"id":1,"name":"multiple"},{"id":2,"name":"Python"},{"id":3,"name":"Unix/Linux/*nix"},{"id":4,"name":"Bash"},{"id":5,"name":"C"},{"id":6,"name":"Sysadmin"},{"id":7,"name":"JavaScript"},{"id":8,"name":"HTML"},{"id":9,"name":"PHP"},{"id":10,"name":"Ruby"},{"id":11,"name":"Aspectwerkz"},{"id":12,"name":"Artificial Intelligence"},{"id":13,"name":"C++"},{"id":14,"name":"MongoDB"},{"id":15,"name":"SQL"},{"id":16,"name":"R"},{"id":17,"name":"GraphQL"},{"id":18,"name":"Swift"},{"id":19,"name":"Android"},{"id":20,"name":"CSS"}],"has_next":true,"has_prev":false,"number_of_pages":4,"page":1,"records_per_page":20,"status":"ok","status_code":200,"total_count":73}
+  /resources/{id}:
+    get:
+      description: Gets more details about a specific resource, given the resource
+        ID
+      parameters:
+      - name: id
+        in: path
+        description: Resource ID
+        required: true
+        style: simple
+        explode: false
+        schema:
+          type: integer
+          format: int64
+      responses:
+        200:
+          description: Resource response successful
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Resource'
+              examples:
+                0:
+                  value: |
+                    {"apiVersion":"1.0","data":{"category":"Books","created_at":"2019-06-15 17:55:32","downvotes":0,"id":3,"languages":[],"last_updated":"","name":"Free Tech Books","notes":"Focuses on general computer science concepts rather than a specific language","paid":false,"times_clicked":0,"upvotes":0,"url":"http://www.freetechbooks.com/"},"status":"ok","status_code":200}
+  /resources:
+    get:
+      description: Gets list of all available resource entries
+      responses:
+        200:
+          description: Resources list retrieval successful
+        404:
+          description: The specified resource was not found
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+    post:
+      description: Creates a new resource
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Resource'
+        required: true
+      responses:
+        200:
+          description: Added new resource
+        404:
+          description: The specified resource was not found
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+      security:
+      - ApiKeyAuth: []
+  /search:
+    get:
+      description: Searches the API for a given keyword
+      responses:
+        200: {}
+  /apikey:
```

Comment:
We definitely need an example of what a proper request looks like for this endpoint